### PR TITLE
Attach summary PDF to tax form Slovensko.sk message

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
Cannot link via development: 

https://github.com/bratislava/private-konto.bratislava.sk/issues/50

### CODE REVIEW

File: nest-forms-backend/src/nases/utils-services/tokens.nases.service.spec.ts [hunk 1]
```diff
@@ -147,17 +150,26 @@ describe('NasesUtilsService', () => {
...
+
+          return readableStream
         })
 
       service['taxService'].getFilledInPdfBase64 = jest
         .fn()
         .mockResolvedValue('AWUKDHLAIUWDHU=====')
 
-      // eslint-disable-next-line xss/no-mixed-html
-      service['stream2buffer'] = jest
```
Comment (MarekBodingerBA) [inline nest-forms-backend/src/nases/utils-services/tokens.nases.service.spec.ts:157]: Removed `stream2buffer` mock as the real implementation in service can be used.

File: nest-forms-backend/src/nases/utils-services/tokens.nases.service.spec.ts [hunk 2]
```diff
@@ -216,9 +228,10 @@ describe('NasesUtilsService', () => {
             `          <RecipientId>example_recipient</RecipientId>\n` +
             `          <MessageType>esmao.eforms.bratislava.obec_024</MessageType>\n` +
             `          <MessageSubject>Podávanie daňového priznanie k dani z nehnuteľností</MessageSubject>\n` +
-            `          <Object Id="id-file0001" IsSigned="false" Name="file0001.pdf" Description="ATTACHMENT" Class="ATTACHMENT" MimeType="application/pdf" Encoding="Base64">TW9jayBmaWxlIGRhdGEgPHRhZz4gc3RyaW5nIDwvdGFnPj4=</Object>\n` +
-            `          <Object Id="id-file0002" IsSigned="false" Name="file0002.pdf" Description="ATTACHMENT" Class="ATTACHMENT" MimeType="application/pdf" Encoding="Base64">TW9jayBmaWxlIGRhdGEgPHRhZz4gc3RyaW5nIDwvdGFnPj4=</Object>\n` +
+            `          <Object Id="id-file0001" IsSigned="false" Name="file0001.pdf" Description="ATTACHMENT" Class="ATTACHMENT" MimeType="application/pdf" Encoding="Base64">dGVzdCBzdHJpbmcgZm9yIGlucHV0OiBwb3NwMDAwMS8xMjM0NTY2Nzg5MDEyMzQ1Njc4OTAvbWluaW9maWxlMDAwMQ==</Object>\n` +
```
Comment (MarekBodingerBA) [inline nest-forms-backend/src/nases/utils-services/tokens.nases.service.spec.ts:231]: These are corrected base64s from stream.

File: nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts [hunk 3]
```diff
@@ -143,14 +143,44 @@ export default class NasesUtilsService {
...
+          _: formPdfBase64,
         })
       } catch (error) {
         console.error(
           `ERROR - Printing form to attachment to Nases and Noris error for form id ${form.id}`,
           error,
         )
       }
+      try {
+        const summaryPdfReadable = await this.convertService.generatePdf(
```
Comment (MarekBodingerBA) [inline nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts:155]: Unfortunately Readable -> Stream -> Base64, but that's how it is implemented.

File: nest-forms-backend/src/convert/convert.service.ts [hunk 4]
```diff
@@ -222,7 +223,10 @@ export default class ConvertService {
       )
     }
 
-    if (isSlovenskoSkTaxFormDefinition(formDefinition)) {
+    if (
+      isSlovenskoSkTaxFormDefinition(formDefinition) &&
+      !forceSummaryForTaxPdf
```
[the remark this answers is not in the document]
Comment (erehulka) [reply nest-forms-backend/src/convert/convert.service.ts:228]: It is not that clear to me either, comment would be helpful.

**Review by erehulka — APPROVED** [review]
Other from the missing comment lgtm

**Pull request merged**